docs-source/content/userguide/managing-domains/fmw-infra/_index.md

```diff
@@ -0,0 +1,353 @@
+  namespace: default
+spec:
+  clusterIP: 10.97.236.215
+  ports:
+  - name: tns
+    port: 1521
+    protocol: TCP
+    targetPort: 1521
+  selector:
+    app.kubernetes.io/instance: dev
+    app.kubernetes.io/name: oracle-db
+  sessionAffinity: None
+  type: ClusterIP
+```
+
+In the example above, the database would be visible in the cluster using the address
+`oracle-db.default.svc.cluster.local:1521/devpdc.k8s`.
+
+When you run the database in the Kubernetes cluster, you will probably want to also
+run RCU from a pod inside your network, though this
+is not strictly necessary.  You could create a `NodePort` to expose your database outside
+the Kubernetes cluster and run RCU on another machine with access to the cluster.
+
+
+##### Running the database outside Kubernetes
+
+If you wish to run the database outside Kubernetes, you need to create a way for containers
+running in pods in Kubernetes to see the database.  This can be done by defining a
+Kubernetes service with no selector and associating it with an endpoint definition, as shown
+in the example below: 
+
+```yaml
+kind: Service
+apiVersion: v1
+metadata:
+ name: database
+spec:
+ type: ClusterIP
+ ports:
+ - port: 1521
+   targetPort: 1521
+---
+kind: Endpoints
+apiVersion: v1
+metadata:
+ name: database
+subsets:
+ - addresses:
+     - ip: 129.123.1.4
+   ports:
+     - port: 1521
+```
+
+This creates a DNS name `database` in the current namespace, or`default` if no namespace is 
+specified, as in the example above.  The fully qualified name would be
+`database.default.svc.cluster.local` in this example.  The second part is the namespace.
+If you looked up the `ClusterIP` for such a service, it would have an IP address on the overlay
+network, that is the network inside the Kubernetes cluster.  If you are using flannel
+for example, the address might be something like `10.0.1.25`.  Note that this is usually a
+non-routed address.
+
+From a container in a pod running in Kubernetes, you can make a connection to that address
+and port `1521`.  Kubernetes will route the connection to the address provided in the
+endpoint definition, in this example `129.123.1.4:1521`.  This IP address (or name) is
+resolved from the point of view of the Kubernetes node's IP stack, not the overlay network
+inside the Kubernetes cluster.  Note that this is a "real" routed IP address. 
+
+When you create your data sources, you would use the internal address, for example
+`database:1521/some.service`.
+
+Because your database is externally accessible, you can run RCU in the normal way, from any
+machine on your network.
+
+#### Running the Repository Creation Utility to set up your database schema
+
+If you want to run RCU from a pod inside the Kubernetes cluster, you can use the Docker
+image that you built earlier as a "service" pod to run RCU.  To do this, start up a
+pod using that image as follows: 
+
+```bash
+kubectl run rcu -ti --image oracle/fmw-infrastructure:12.2.1.3 -- sleep 100000
```

Comment:
I googled "how to create a container that runs forever," and found this: 

`trap : TERM INT; (while true; do sleep 1000; done) & wait`

... might be better than just a really long sleep.

Reply:
Going to ignore this for now.